### PR TITLE
Notification service

### LINKED ISSUE #15: Notification service implementation

### PR DESCRIPTION
**Pull request by harshadk-sourcefuse**
## Description
Notification Service implementation

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Intermediate change (work in progress)

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

## Checklist:

- [ ] Performed a self-review of my own code
- [X] npm test passes on your machine
- [X] New tests added or existing tests modified to cover all changes
- [ ] Code conforms with the style guide
- [ ] API Documentation in code was updated
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: services/notification-service/src/main/java/com/sourcefuse/jarc/services/notificationservice/NotificationServiceApplication.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.annotation.web.configurers.CsrfConfigurer;
+import org.springframework.security.web.SecurityFilterChain;
+
+@SpringBootApplication
+@ComponentScan(
+  {
+    "com.sourcefuse.jarc.core",
+    "com.sourcefuse.jarc.services.notificationservice"
```
Comment (mlakdawala-sf) [inline services/notification-service/src/main/java/com/sourcefuse/jarc/services/notificationservice/NotificationServiceApplication.java:15]: please add authlib package to use auth middlewares

File: services/notification-service/src/test/java/com/sourcefuse/jarc/services/notificationservice/GroupControllerTest.java [hunk 2]
```diff
@@ -0,0 +1,269 @@
...
+//            .andExpect(MockMvcResultMatchers.jsonPath("$.name").value(savedGroup.getName()));
+//​
+//    // Verify the expected interactions and assertions
+//    Mockito.verify(groupRepository, Mockito.times(1)).save(ArgumentMatchers.any(Group.class));
+//    Mockito.verify(userGroupsRepository, Mockito.times(1)).save(ArgumentMatchers.any(UserGroup.class));
+//  }
+//​
+//  private static void extracted() {
+//    UsernamePasswordAuthenticationToken auth =
+//      new UsernamePasswordAuthenticationToken(
```
Comment (mlakdawala-sf) [inline services/notification-service/src/test/java/com/sourcefuse/jarc/services/notificationservice/GroupControllerTest.java]: can we remove this file

**Review by mlakdawala-sf — CHANGES_REQUESTED** [review]
Rest looks good to me apart from sonar

File: services/notification-service/src/test/java/com/sourcefuse/jarc/services/notificationservice/GroupControllerTest.java (hunk 2 above)
Comment (harshadk-sourcefuse) [reply services/notification-service/src/test/java/com/sourcefuse/jarc/services/notificationservice/GroupControllerTest.java]: Done

File: services/notification-service/src/main/java/com/sourcefuse/jarc/services/notificationservice/NotificationServiceApplication.java (hunk 1 above)
Comment (harshadk-sourcefuse) [reply services/notification-service/src/main/java/com/sourcefuse/jarc/services/notificationservice/NotificationServiceApplication.java:15]: will do in separate PR

**Review by mlakdawala-sf — APPROVED** [review]

**Pull request merged**